exploration/text-vs-code.md

```diff
@@ -0,0 +1,216 @@
+the primarily unquoted translatable message contents.
+
+For messages with multiple variants,
+separate the variants using `when` statements.
+
+Trim whitespace between and around statements such as `input` and `when`,
+but do not otherwise trim any leading or trailing whitespace from a message.
+This allows for whitespace such as spaces and newlines to be used outside patterns
+to make a message more readable.
+
+Allow for a pattern to be `{{…}}` quoted
+such that it preserves its leading and/or trailing whitespace
+even when preceded or followed by statements.
+
+## Alternatives Considered
+
+### Start in code, encapsulate text
+
+This approach treats messages as something like a resource format for pattern values.
+Keywords are declared directly at the top level of a message,
+and patterns are always surrounded by `{{…}}` or some other delimiters.
+
+Whitespace in patterns is never trimmed.
+
+The `{{…}}` are required for all messages,
+including ones that only consist of text.
+Delimiters of the resource format are required in addition to this,
+so messages may appear wrapped as e.g. `"{{…}}"`.
+
+This option is not chosen due to adding an excessive
+quoting burden on all messages.
+
+### Start in text, encapsulate code, re-encapsulate text within code
+
+As in the proposed design, simple patterns are unquoted.
+Patterns in messages with statements, however,
+are required to always be surrounded by `{{…}}` or some other delimiters.
+
+This effectively means that some syntax will "enable" code mode for a message,
+and that patterns in such a message need delimiters.
+
+This option is not chosen due to adding an excessive
+quoting burden on all multi-variant messages,
+as well as introducing an unnecessary additional conceptual layer to the syntax.
+
+### Start in text, encapsulate code, trim minimally
+
+This is the same as the proposed design,
+but with a different trimming rule:
+
+- Trim all spaces before and between declarations.
+- For single-variant messages, trim one newline after the last declaration.
+- For multivariant messages,
+  trim one space after a `when` statement and
+  one newline followed by any spaces before a subsequent `when` statement.
+
+This option is not chosen due to the quoting being too magical.
+Even though this allows for all patterns with whitespace to not need quotes,
+the cost in complexity is too great.
+
+### Start in text, encapsulate code, trim maximally
+
+This is the same as the proposed design,
+but with a different trimming rule:
+
+- Trim all leading and trailing whitespace for each pattern.
+
+Expressing the trimming on patterns rather than statements
+means that leading and trailing spaces are also trimmed from simple messages.
+This option is not chosen due to this being somewhat surprising,
+especially when messages are embedded in host formats that have predefined means
+of escaping and/or trimming leading and trailing spaces from a value.
```

Comment:
Trimming simple patterns like this is a bridge too far for me. It should be a separate decision for the "trim XXX" options whether they are trimmed. I can make a plausible argument for why simple patterns should behave differently when trimmed than variant patterns do.